### ISSUE TITLE
heap arenas/bins broken

### ISSUE THREAD

**Issue by Grazfather** [hugsy/gef#97]
probably exists in other places. I don't fully understand what's happening, but it's because of {:s} in `GlibcHeapArenaCommand`. Changing to `{}` fixes the issue.

This is on gdb with python 3, so it probably has something to do with a string being converted to `bytes` at some point. See [here](http://stackoverflow.com/questions/24170519/python-typeerror-non-empty-format-string-passed-to-object-format#24170567)

```python
gef➤  heap arenas
[+] Listing active arena(s):
Python Exception <class 'TypeError'> non-empty format string passed to object.__format__:
Error occurred in Python command: non-empty format string passed to object.__format__
```

**Comment by Grazfather** [comment]
Heaps bins broken also. piggybacking on this issue.

Running `heap bins` on a fresh heap causes an issue when the unsorted bin is printed (since it reads the address as 0 and tries to dereference).

```python
───────────────────────────────────────[ Unsorted Bin for arena 'main_arena' ]───────────────────────────────────────
[+] Found base for bin(0): fw=0x0, bk=0x0
0x7ffff7dd3760
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x10:
Python Exception <class 'gdb.error'> Error occurred in Python command: Cannot access memory at address 0x10:
Error occurred in Python command: Error occurred in Python command: Cannot access memory at address 0x10
```

**Comment by Grazfather** [comment]
Subsequent bins would be broken as well, we just don't get that far.

I have no idea how `self.bins` works, since I don't see it assigned anywhere. We can probably get around this issue simply by checking if an address is mapped (slow) before dereferencing it, or just catch and ignore the exception when it occurs (preferred)

**Comment by hugsy** [comment]
Man, I haven't retested all the features since the move to `str.format`... So many regressions, that's not acceptable... I need to work on unit tests.
Anyway, I'll get a look and fix this `heap` command issue.
Thanks for reporting.

**Comment by hugsy** [comment]
Should be fixed by 01902a8.

Can you confirm?

**Comment by Grazfather** [comment]
Yeah avoids the exception prints.
